cmd/svcat/broker/register_cmd.go

```diff
@@ -18,57 +18,122 @@ package broker
+	URL               string
 }
 
 // NewRegisterCmd builds a "svcat register" command
 func NewRegisterCmd(cxt *command.Context) *cobra.Command {
 	registerCmd := &RegisterCmd{
-		Context: cxt,
+		Context:  cxt,
+		Waitable: command.NewWaitable(),
 	}
 	cmd := &cobra.Command{
-		Use:   "register NAME --url URL",
+		Use:   "register NAME URL",
 		Short: "Registers a new broker with service catalog",
 		Example: command.NormalizeExamples(`
-		svcat register mysqlbroker --url http://mysqlbroker.com
+		svcat register mysqlbroker http://mysqlbroker.com
 		`),
 		PreRunE: command.PreRunE(registerCmd),
 		RunE:    command.RunE(registerCmd),
 	}
-	cmd.Flags().StringVar(&registerCmd.URL, "url", "",
-		"The broker URL (Required)")
-	cmd.MarkFlagRequired("url")
+	cmd.Flags().StringVar(&registerCmd.BasicSecret, "basic-secret", "",
+		"A secret containing basic auth (username/password) information to connect to the broker")
+	cmd.Flags().StringVar(&registerCmd.BearerSecret, "bearer-secret", "",
+		"A secret containing a bearer token to connect to the broker")
+	cmd.Flags().StringVar(&registerCmd.CAFile, "ca", "",
+		"A file containing the CA certificate to connect to the broker")
+	cmd.Flags().StringSliceVar(&registerCmd.ClassRestrictions, "class-restrictions", []string{},
+		"A whitelist of Classes to allow from the broker")
+	cmd.Flags().StringSliceVar(&registerCmd.PlanRestrictions, "plan-restrictions", []string{},
+		"A whitelist of Plans to allow from the broker")
+	cmd.Flags().StringVar(&registerCmd.RelistBehavior, "relist-behavior", "",
+		"Behavior for relisting the broker's catalogue. Manual/Duration")
+	cmd.Flags().DurationVar(&registerCmd.RelistDuration, "relist-duration", 0*time.Second,
+		"Interval to refetch broker catalog when relist-behavior is set to duration, specified in human readable format: 30s, 1m, 1h")
+	cmd.Flags().BoolVar(&registerCmd.SkipTLS, "skip-tls", false,
+		"Disables TLS certificate verification when communicating with this broker. This is strongly discouraged. You should use --ca instead.")
+	registerCmd.AddWaitFlags(cmd)
+
 	return cmd
 }
 
 // Validate checks that the required arguements have been provided
 func (c *RegisterCmd) Validate(args []string) error {
-	if len(args) == 0 {
-		return fmt.Errorf("a broker name is required")
+	if len(args) < 2 {
+		return fmt.Errorf("a broker name and url is required")
 	}
 	c.BrokerName = args[0]
+	c.URL = args[1]
 
+	if c.BasicSecret != "" && c.BearerSecret != "" {
+		return fmt.Errorf("cannot use both basic auth and bearer auth")
+	}
+
+	if c.CAFile != "" {
+		_, err := os.Stat(c.CAFile)
+		if err != nil {
+			return fmt.Errorf("error finding CA file: %v", err.Error())
+		}
+	}
+	if c.RelistBehavior != "" {
+		c.RelistBehavior = strings.ToLower(c.RelistBehavior)
+		if c.RelistBehavior != "duration" && c.RelistBehavior != "manual" {
+			return fmt.Errorf("relist-behavior must be duration/manual")
+		}
+	}
+	if c.RelistDuration != 0*time.Second && c.RelistBehavior != "duration" {
```

Comment:
Is this something that the api server would accept? I am wondering if we are applying validations on the cli to values that the server would allow. Can you check please?

Reply:
api server drops the relist duration on the ground. I guess we can remove this.